aptos-move/aptos-vm/src/move_vm_ext/session/mod.rs

```diff
@@ -556,6 +556,10 @@ where
     let session_counter = session_id.session_counter();
     let txn_hash = session_id.txn_hash();
 
+    // Note: if any new native functions that return references are added,
```

Comment:
An idea: we can have a marker trait so that every new context has to implement this and hence - register the functions if they return native (and compiler complaint is much stronger than a comment).

One issue is if natives are added, they might be added to the same context, so we cannot enforce this on per-function level :/